### ISSUE TITLE
Example for 3D convolutions

### ISSUE THREAD

**Issue by m-albert** [getkeops/keops#36]
Thanks a lot for this useful code, the benchmarks are impressive.

I wanted to try it out for convolutions (anisotropic kernels) on gridded data (3D images). Could you point me to an example code for this use case (couldn't find it)?

That'd be great, thanks a lot in advance!

**Comment by jeanfeydy** [comment]
Hi @m-albert ,

You're welcome!
Would [this](http://www.kernel-operations.io/keops/_auto_examples/pytorch/plot_anisotropic_kernels.html), [this](http://www.kernel-operations.io/keops/_auto_tutorials/gaussian_mixture/plot_gaussian_mixture.html) or [this](http://www.kernel-operations.io/keops/_auto_tutorials/interpolation/plot_RBF_interpolation_torch.html) tutorial fit with what you had in mind? 
Please note that the KeOps routines serve a very different purpose from the standard "Conv" layers of PyTorch or TensorFlow: positions should be encoded explicitly using `x` and `y` coordinates. They are perfectly suited to situations:
- where you're only interested in computing convolutions on small domains (e.g. thin segmentation masks in a full volume), 
- where you're extrapolating some off-grid signal onto a full grid (e.g. a GMM log-likelihood, as in the tutorials above).

However, if your goal is to compute good old convolutions on "full" 3D volumes, standard PyTorch/TF layers are still your best bet I'm afraid :-)
What do you think?

Best regards,
Jean

**Comment by m-albert** [comment]
@jeanfeydy thanks for your quick response and the pinpointing!

And thanks for clarifying: Indeed I thought there would be a convenient way of expressing "good old convolutions" as reductions that would fit the KeOps framework.

I was speaking to @bcharlier last Friday and thought to remember you had a way of performing tiled grid convolutions on the GPU.

Cheers,
Marvin

**Comment by jeanfeydy** [comment]
I see: good luck, then!
